addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1855,17 +1878,17 @@ var SnippetsMenu = Widget.extend({
      * @param {OdooEvent} ev
      */
     _onUpdateCustomizeElements: function (ev) {
-        while (this.customizePanel.firstChild) {
-            this.customizePanel.removeChild(this.customizePanel.firstChild);
-        }
```

Comment:
I chose to trust this when I made my original code: https://jsperf.com/innerhtml-vs-removechild/37
Also some people claims that innerHTML might lead to memory leak in some browsers (no proof though)

Reply:
So, looping to remove is about 5 times faster than innerHTML. That's great but innerHTML still only takes 109ns, so 0.0001ms and we're toggling classes right after which will cause a reflow anyway so any performance penalty here is dwarfed by that.

As far as memory leaks, I found one reference on inserting nested tables with innerHTML causing leaks in ie 9-11, but that shouldn't be a problem here.

One problem that could cause a leak is if there is jQuery data still attached to the elements (although jQuery now uses the native HTML5 `dataset`, which should be cleaned up by the browser on removal, but maybe there is a fallback for older browsers), but in that case, looping would cause the same leak and we should really be using `$(this.customizePanel).empty()` instead, which IMO is also the most meaningful, but if we're trying to get away from using jQuerry, `innerHTML = ""` seems like a good option.

Reply:
Of course we don't care about performance here (even if personally, I don't see a reason to choose a "less performant" option when there is already the good one in place). But the point was: why changing code that does not need to be changed? :wink: Anyways, I am fine with it now that it is done but don't forget that changing code to do the same thing adds noise in the PRs JKE and I review (I did lose time searching if there is a reason you changed that).

> although jQuery now uses the native HTML5 dataset, which should be cleaned up by the browser on removal, but maybe there is a fallback for older browsers

Mmh, strange, RDE had problem earlier this week that suggests that jQuery don't use `dataset` (since the solution was to use `dataset` instead). Anyways, clearly, better to use `dataset` directly when we can.

Reply:
> Mmh, strange, RDE had problem earlier this week that suggests that jQuery don't use `dataset`

I thought I had read that somewhere, I misunderstood. jQuery now pulls the dataset attribute into its own data when it is constructed, so if a node already has a dataset, you can access it with .data()